qemu/tests/qemu_guest_agent.py

```diff
@@ -488,23 +750,17 @@ class QemuGuestAgentBasicCheckWin(QemuGuestAgentBasicCheck):
     """
     @error.context_aware
     def setup_gagent_in_host(self, params, vm):
-        error.context("Install qemu guest agent package on host", logging.info)
-        gagent_host_install_cmd = params["gagent_host_install_cmd"]
-        utils.run(gagent_host_install_cmd,
+        error.context("download qemu-ga.msi to host", logging.info)
+        gagent_download_cmd = params["gagent_download_cmd"]
+        utils.run(gagent_download_cmd,
                   float(params.get("login_timeout", 360)))
-
-        error.context("Install dependence packages on host", logging.info)
-        gagent_host_dep_install_cmd = params.get("gagent_host_dep_install_cmd",
-                                                 "")
-        utils.run(gagent_host_dep_install_cmd,
-                  float(params.get("login_timeout", 360)))
-
-        error.context("Copy necessary DLLs to guest", logging.info)
+        gagent_host_path = params["gagent_host_path"]
+        if not os.path.exists(gagent_host_path):
+            raise error.TestFail("qemu-ga install program is not exist, maybe "
+                                 "the program is not successfully downloaded ")
+        session = self._get_session(params, vm)
         gagent_guest_dir = params["gagent_guest_dir"]
         gagent_remove_service_cmd = params["gagent_remove_service_cmd"]
-        gagent_dep_dlls_list = params.get("gagent_dep_dlls", "").split()
-
-        session = self._get_session(params, vm)
         s, o = session.cmd_status_output("mkdir %s" % gagent_guest_dir)
```

Comment:
Under Windows os, I didn't find the "-p" option for command "mkdir " linke UNIX-like os.  So I do it in another way. It will handle the result if the return value of "mkdir" is "1", this means the directory is existed, then it will be processed.